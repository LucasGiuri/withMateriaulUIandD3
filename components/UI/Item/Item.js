import { ItemContainer } from './Item.styles';
import PropTypes from 'prop-types';

const Item = ({children}) => <ItemContainer>{children}</ItemContainer>;

Item.propTypes = {
  children: PropTypes.node
};

export default Item;