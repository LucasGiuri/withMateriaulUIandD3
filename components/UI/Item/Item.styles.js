import styled from 'styled-components';
import { WHITE, OLD_GREY } from '../../../public/styles/colors';

export const ItemContainer = styled.div`
  padding: 14px;
  display: flex;
  flex-direction: row;
  cursor: pointer;
  border-bottom: 1px solid ${OLD_GREY};
  font-family: 'Roboto', sans-serif;
  font-size: 22px;
  transition: 0.25s;
  color: ${OLD_GREY};
  &:hover {
    color: ${WHITE};
  }
`;