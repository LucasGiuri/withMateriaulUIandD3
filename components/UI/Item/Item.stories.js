import React from 'react';
import { storiesOf } from '@storybook/react';
import Item from './Item';

storiesOf('Item')
  .add('Item default', () => (
    <Item><h2>Item</h2></Item>
  ))
