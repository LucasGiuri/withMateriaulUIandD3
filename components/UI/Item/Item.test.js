import React from 'react';
import renderer from 'react-test-renderer';
import Item from './Item';

describe('Item', () => {
  it('Item basic render snapshoot', () => {
    const children = <p>Link to google</p>;
    const component = renderer.create(<Item>{children}</Item>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
