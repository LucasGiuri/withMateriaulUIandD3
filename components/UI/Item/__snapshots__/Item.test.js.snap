// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Item basic render snapshoot 1`] = `
<div
  className="Itemstyles__ItemContainer-sc-18euzxw-0 iNynqh"
>
  <p>
    Link to google
  </p>
</div>
`;
